core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -779,6 +779,25 @@ public final String getUserClaimValue(String userName, String claim, String prof
             return (String[]) object;
         }
 
+        if (log.isDebugEnabled()) {
+            log.debug("Listing users who having value as " + claimValue + " for the claim " + claim);
+        }
+
+        if (USERNAME_CLAIM_URI.equalsIgnoreCase(claim)) {
+
+            if (log.isDebugEnabled()) {
+                log.debug("Switching to list users using username");
+            }
+
+            String[] filteredUsers = listUsers(claimValue, -1);
```

Comment:
would it be good to derive this -1 as a constant that describes what this value is ?